### ISSUE TITLE
[Bug] Registering multiple named instances of this module fail.

### ISSUE THREAD

**Issue by darkgl0w** [fastify/fastify#3323]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.21.2

### Plugin version

3.6.0

### Node.js version

14.17.6

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

ArchLinux (kernel 5.14.5-arch1-1)

### Description

Since https://github.com/fastify/fastify/pull/3317 was merged in Fastify it's impossible to register multiple named instances of the package.

This error is thrown at the start:
```
[...]/node_modules/fastify/lib/decorate.js:40
    throw new FST_ERR_DEC_ALREADY_PRESENT(name)
          ^
FastifyError [Error]: The decorator 'pg' has already been added! 
[...]
```
I am currently investigating the code to clearly identify the faulty part and submit a fix PR.
I suspect that this is due to https://github.com/fastify/fastify-postgres/pull/76 as the code rely on a mutable request.pg object (https://www.fastify.io/docs/latest/Decorators/#decoraterequestname-value-dependencies).

I think we should adopt the same request decoration strategy as the one used for `fastify-jwt` (https://github.com/fastify/fastify-jwt/pull/175).

WDYT can I go for a fix PR adopting this decoration strategy knowing that this will be a breaking change ?

### Steps to Reproduce

```js
'use strict';

const fastify = require('fastify');

const server = fastify();

server.register(require('fastify-postgres'), {
  connectionString: 'postgres://postgres:postgres@localhost/postgres',
  name: 'one'
})
server.register(require('fastify-postgres'), {
  connectionString: 'postgres://postgres:postgres@localhost/postgres',
  name: 'two'
})

server.listen(3000);
```

### Expected Behavior

It should start.

**Comment by darkgl0w** [comment]
I have tested if `fastify-jwt` crashes too when registering 2 named instances and it run just like it should.
I even have a fix here https://github.com/darkgl0w/fastify-postgres/tree/fix-decorator-bug for this package if you want to look at it before I submit a pull request.

**Comment by climba03003** [comment]
> 
> 
> I have tested if `fastify-jwt` crashes too when registering 2 named instances and it run just like it should.
> I even have a fix here https://github.com/darkgl0w/fastify-postgres/tree/fix-decorator-bug for this package if you want to look at it before I submit a pull request.

As I mention, it is a bug from `fastify`. The fix should be inside the main repo.

**Comment by mcollina** [comment]
@climba03003 do you understand why this is happening? Could you send a PR to fix? I can ship a release if the code is there.

**Comment by darkgl0w** [comment]
ok, that's nice if you are already working on a fix ^^
Ping me if you need some help :)

PS: sorry I read your previous message too fast and missed the point of it :d

**Comment by mcollina** [comment]
Fixed in v3.21.3.
